subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformationRegistrationFactory.java

```diff
@@ -144,6 +144,7 @@ public ArtifactTransformRegistration create(ImmutableAttributes from, ImmutableA
                 String.format(formatString, ModelType.of(implementation).getDisplayName()),
                 validationMessages.keySet().stream()
                     .sorted()
+                    .distinct()
```

Comment:
`distinct()` doesn't make much sense here, since we are looking at the `keySet()` of a map, right?

Reply:
Good point, removed.